Editor/CourseValidation/ValidationHandler.cs

```diff
@@ -29,39 +36,55 @@ public ValidationHandler()
-        /// <param name="entity">Object, which will be validated.</param>
-        /// <param name="course">Course where given <paramref name="entity"/> belongs.</param>
+        /// <param name="data">Data object, which will be validated.</param>
+        /// <param name="course">Course where given <paramref name="data"/> belongs.</param>
         /// <param name="context">Context of the validation.</param>
-        /// <returns>List of reports regarding invalid objects related to the <paramref name="entity"/>.</returns>
-        public ValidationReport Validate(IEntity entity, ICourse course, IContext context = null)
+        /// <returns>List of reports regarding invalid objects related to the <paramref name="data"/>.</returns>
+        public ValidationReport Validate(IData data, ICourse course, IContext context = null)
         {
             Stopwatch stopwatch = Stopwatch.StartNew();
 
             if (context == null)
             {
-                context = ContextResolver.FindContext(entity, course);
+                context = ContextResolver.FindContext(data, course);
             }
-            List<EditorReportEntry> entries = InternalValidate(entity, context);
+            List<EditorReportEntry> entries = InternalValidate(data, context);
 
             stopwatch.Stop();
-            return new ValidationReport(entries, stopwatch.ElapsedMilliseconds);
+            if (LastReport == null)
+            {
+                LastReport = new ValidationReport(entries, stopwatch.ElapsedMilliseconds);
+            }
+            else
+            {
+                LastReport.Update(entries, context, stopwatch.ElapsedMilliseconds);
+            }
+            return LastReport;
+        }
+
+        /// <summary>
+        /// Checks if validation is currently allowed.
+        /// </summary>
+        public bool IsAllowedToValidate()
+        {
+            return CreatorProjectSettings.Load().IsValidationEnabled && RuntimeConfigurator.Exists;
         }
 
         /// <summary>
-        /// Calls internal validation process for given <paramref name="entityObject"/>.
+        /// Calls internal validation process for given <paramref name="validateableObject"/>.
         /// </summary>
-        /// <param name="entityObject">Object which is the target of the validation.</param>
+        /// <param name="validateableObject">Object which is the target of the validation.</param>
         /// <param name="context">Context this validation runs in, has to be the correct one.</param>
-        /// <returns>List of reports regarding invalid objects related to the <paramref name="entityObject"/>.</returns>
-        protected List<EditorReportEntry> InternalValidate(object entityObject, IContext context)
+        /// <returns>List of reports regarding invalid objects related to the <paramref name="validateableObject"/>.</returns>
+        protected List<EditorReportEntry> InternalValidate(object validateableObject, IContext context)
```

Comment:
```suggestion
        protected List<EditorReportEntry> InternalValidate(object validatableObject, IContext context)
```
It is actually _validatable_ without the 'e'